oxide/lib_test.go

```diff
@@ -127,9 +127,16 @@ func Test_buildRequest(t *testing.T) {
 		//			wantErr: "Some error that doesn't exist yet",
 		//		},
 	}
+
+	// Just to get a client to call buildRequest on.
+	c, err := NewClientWithOpts()
+	if err != nil {
+		t.Fatalf("failed creating api client: %v", err)
+	}
+
```

Comment:
It'd be really nice to some tests that make sure the client is built with the options

Reply:
This is still pending. I figured we could finalize the design first before updating the tests.

Reply:
I haven't forgotten about these tests. I pushed a new commit addressing the latest feedback. I'm used to pushing new amended commits, but I realized after that you may have preferred separate commits for review purposes. I apologize if I messed up your flow with that. Let me know what you think about the latest changes and then I'll tackle the testing aspect.

Reply:
> but I realized after that you may have preferred separate commits for review purposes

hahaha yeah, no worries though :)

Reply:
Resolving this in favor of the discussion happening in the test file.